examples/todo-typescript/src/client/MainPage.tsx

```diff
@@ -0,0 +1,94 @@
+import './Main.css';
+import React from 'react';
+import logout from '@wasp/auth/logout.js';
+import useAuth from '@wasp/auth/useAuth.js';
+import { useQuery } from '@wasp/queries'; // Wasp uses a thin wrapper around react-query
+import getTasks from '@wasp/queries/getTasks';
+import createTask from '@wasp/actions/createTask';
+import updateTask from '@wasp/actions/updateTask';
+import waspLogo from './waspLogo.png';
+import { Task } from './types'
+
+const MainPage = () => {
+  const { data: user } = useAuth();
+  const { data: tasks, isLoading, error } = useQuery(getTasks);
+
+  React.useEffect(() => {
+    console.log(user);
+  }, [user]);
+
+  if (isLoading) return 'Loading...';
+  if (error) return 'Error: ' + error;
+
+  return (
+    <main>
+      <img src={waspLogo} alt='wasp logo' />
+      <h1>
+        {user.username}
+        {`'s tasks :)`}
+      </h1>
+      <NewTaskForm />
+      {tasks && <TasksList tasks={tasks} /> }
+      <button onClick={logout}> Logout </button>
+    </main>
+  );
+};
+
+const Todo = ({task, number}: {task: Task, number: number}) => { 
+  const handleIsDoneChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    try {
+      await updateTask({
+        taskId: task.id,
+        isDone: event.currentTarget.checked,
+      });
+    } catch (err: any) {
+      window.alert('Error while updating task: ' + err?.message);
+    }
+  };
+
+  return (
+    <div>
+      <span>
+        {number + 1}
+        {''}
+      </span>
+      <input type='checkbox' id={task.id.toString()} checked={task.isDone} onChange={handleIsDoneChange} />
+      <span>{task.description}</span>{' '}
+    </div>
+  );
+};
+
+const TasksList = ({tasks}: { tasks: Task[] }) => {
+  if (tasks.length === 0) return <p>No tasks yet.</p>;
+  return (
+    <div className='tasklist'>
+      {tasks.map((tsk, idx) => (
+        <Todo task={tsk} number={idx} key={idx} />
+      ))}
+    </div>
+  );
+};
```

Comment:
I can recommend several things here.

### Numbering the list
You can skip sending the `number` prop if you turn this into an ordered list:
  - Change `<div className='tasklist'>` into `<ol className='tasklist'>`
  - Change the `div` inside `Todo` into an `li` (you can drop the `span` element that displays the number)
  
If you knew this but decided against it for styling reasons, ignore my comment :)

### Spreading object props
Assuming you drop the `number`, the `Todo` becomes a display model for `Task`. You can change it to directly accept the props you need inside the component, saving you from having to access `task.something` multiple times. 

To elaborate, instead of:
```typescript
const Todo = ({ task }: { task: Task }) => {
  // And having to access task.id, task.description, task.isDone...
}
```
You can do:
```typescript
const Todo = ({ id, isDone, description }: Task) => {
  // And directly access id, isDone, description
}
```
Don't worry, it won't make sending props more verbose. Just turn this:
```tsx
<Task task={task} key={idx} />
```
Into this:
```tsx
<Task {...task} key={idx} />
```